### PR TITLE
Add custom HTTP server response files

### PR DESCRIPTION
**Pull request by 6ixfalls**
## Description
Allows for a custom file to be used in case of a 404 error and to replace the default "Welcome to Grasscutter" message.
> Also fixed URLDecoder IntelliSense warning

## Type of changes

- [ ] Bug fix
- [X] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist

- [X] My code follows the style guidelines of this project
- [X] My pull request is unique and no other pull requests have been opened for these changes
- [X] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [X] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by memetrollsXD** [comment]
For what would this be useful for?

**Comment by 6ixfalls** [comment]
Adding a custom response file that doesn't show the default "Welcome" page or 404 page? Nice for customization imo

**Comment by 6ixfalls** [comment]
Might be nice to consolidate all of these commits but there's so many refactors to DispatchServer

**Comment by KingRainbow44** [comment]
the pain of having to merge other dispatch conflicts into my pr
*someone help me I'm dying*

File: src/main/java/emu/grasscutter/utils/ConfigContainer.java [hunk 1]
```diff
@@ -120,6 +120,9 @@ public static class Dispatch {
         public int bindPort = 443;
         /* This is the port used in URLs. */
         public int accessPort = 0;
+        
+        public String WelcomeFile = "./welcome.html";
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/utils/ConfigContainer.java:124]: Use `lowerCamelCase` in the config file please.

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

**Comment by 4Benj** [comment]
Due to a pending dispatch rework by @KingRainbow44, I will be closing this PR to prevent any merge conflicts. Hopefully this will be included in the rework but if not feel free to reopen this PR or open a new one
If it wasn't for my idiocy this would've been merged by now, I'm very sorry. 

Thank you for contributing to Grasscutter

**Pull request closed without merging**